File: ebordo_mobile/lib/models/izvjestaj.dart
import 'dart:convert';
import 'package:ebordo_mobile/models/utakmica-izmjena.dart';
import 'package:ebordo_mobile/models/utakmica-nastup.dart';
import 'package:ebordo_mobile/models/utakmica.dart';
import 'igrac.dart';

class Izvjestaj {
  final int izvjestajId;
  final num goloviSarajevo;
  final num goloviProtivnik;
  final int rezultat;
  final String datumKreiranja;
  final String zapisnik;
  final int vrijeme;
  final Igrac igracUtakmica;
  final Utakmica utakmica;
  final List<int> slikaSaUtakmice;
  final List<UtakmicaIzmjena> izmjene;
  final List<UtakmicaNastup> nastupi;

  Izvjestaj({
    required this.izvjestajId,
    required this.goloviSarajevo,
    required this.goloviProtivnik,
    required this.rezultat,
    required this.datumKreiranja,
    required this.zapisnik,
    required this.vrijeme,
    required this.igracUtakmica,
    required this.utakmica,
    required this.slikaSaUtakmice,
    required this.izmjene,
    required this.nastupi,
  });

  factory Izvjestaj.fromJson(Map<String, dynamic> json) {
    String stringByte = json["slikaSaUtakmice"] as String;
    List<int> bytes = base64.decode(stringByte);
    return Izvjestaj(
        izvjestajId: int.parse(json["izvještajId"].toString()),
        goloviSarajevo: json["goloviSarajevo"],
        goloviProtivnik: json["goloviProtivnik"],
        rezultat: json["rezultat"],
        datumKreiranja: json["datumKreiranja"],
        zapisnik: json["zapisnik"],
        vrijeme: json["vrijeme"],
        igracUtakmica: Igrac.fromJson(json['igracUtakmica']),
        utakmica: Utakmica.fromJson(json['utakmica']),
        slikaSaUtakmice: bytes,
        izmjene: (json['izmjene'] as List)
            .map((i) => UtakmicaIzmjena.fromJson(i))
            .toList(),
        nastupi: (json['nastupi'] as List)
            .map((i) => UtakmicaNastup.fromJson(i))
            .toList());
  }

  Map<String, dynamic> toJson() => {
        "izvjestajId": izvjestajId,
        "goloviSarajevo": goloviSarajevo,
        "goloviProtivnik": goloviProtivnik,
        "rezultat": rezultat,
        "zapisnik": zapisnik,
        "vrijeme": vrijeme,
        "igracUtakmica": igracUtakmica,
        "utakmica": utakmica,
        "slikaSaUtakmice": slikaSaUtakmice,
        "izmjene": izmjene,
        "nastupi": nastupi,
      };
}

File: eBordo.Model/Models/Izvjestaj.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eBordo.Model.Models
{
    public class Izvjestaj
    {
        public int izvjestajId { get; set; }
        public int goloviSarajevo { get; set; }
        public int goloviProtivnik { get; set; }
        public Rezultat rezultat { get; set; }
        public DateTime datumKreiranja { get; set; }
        public string zapisnik { get; set; }
        public byte[] slikaSaUtakmice { get; set; }
        public Vrijeme vrijeme { get; set; }

        public ICollection<UtakmicaNastup> nastupi { get; set; }
        public ICollection<UtakmicaIzmjena> izmjene { get; set; }

        public Igrac igracUtakmica { get; set; }
        public int igracUtakmicaID { get; set; }

        public Utakmica utakmica { get; set; }
        public int utakmicaID { get; set; }

        public Trener trener { get; set; }
        public int trenerID { get; set; }
    }
}
